packages/teleport/src/Main/useMain.ts

```diff
@@ -16,11 +16,14 @@ limitations under the License.
 
 import { useState } from 'react';
 import useAttempt from 'shared/hooks/useAttemptNext';
-import useTeleport from 'teleport/useTeleport';
 import { Feature } from 'teleport/types';
+import teleportContext from 'teleport/teleportContext';
+import teleportEContext from 'e-teleport/teleportEContext';
 
-export default function useMain(features: Feature[]) {
-  const ctx = useTeleport();
+export default function useMain(
+  features: Feature[],
+  ctx: teleportContext | teleportEContext
```

Comment:
`ctx: teleportContext | teleportEContext`

why? Assuming that E version would be extending the OSS it would still work

```tsx
class TeleportContext extends OSSContext {
}
```

Reply:
revised